src/as_execution/abi.rs

```diff
@@ -1110,7 +1118,7 @@ fn read_string(memory: &Memory, store: &impl AsStoreRef, ptr: i32) -> ABIResult<
 
 /// Tooling, return a pointer offset of a serialized list in json
 fn alloc_string_array(ctx: &mut FunctionEnvMut<ASEnv>, vec: &[String]) -> ABIResult<i32> {
-    let env = ctx.data().clone();
+    let env = get_env(&ctx)?;
     let addresses = serde_json::to_string(vec)?;
     Ok(StringPtr::alloc(&addresses, env.get_ffi_env(), ctx)?.offset() as i32)
 }
```

Comment:
I can't comment above but on L.1164 there is still a `ctx.data().clone()` that should be replaced